### PR TITLE
Proper fstat implementation

### PR DESCRIPTION
**Pull request by mcpherrinm**
Before, calling fstat always returned 0644.

Now, it fstat should match stat.

TODO: Tests.

### CODE REVIEW

File: fs.go [hunk 1]
```diff
@@ -170,41 +170,50 @@ func (kwfs KeywhizFs) Open(name string, flags uint32, context *fuse.Context) (no
 	kwfs.Debugf("Open called with '%v'", name)
 
 	var file nodefs.File
+	var mode uint32
 	switch {
 	case name == "", name == ".json", name == ".json/secret":
 		return nil, fuseEISDIR
 	case name == ".version":
 		file = nodefs.NewDataFile([]byte(fsVersion))
+		mode = 0440
```
Comment (alokmenghrajani) [inline fs.go]: why not 0444 for version?

**Comment by alokmenghrajani** [comment]
👍

**Comment by csstaub** [comment]
Tests plz

File: fs.go (hunk 1 above)
Comment (mcpherrinm) [inline fs.go]: fixed

**Comment by mcpherrinm** [comment]
Added tests

**Comment by csstaub** [comment]
LGTM 🚢

**Pull request merged**